.pre-commit-config.yaml

```diff
@@ -1,8 +1,92 @@
 repos:
-  - repo: https://github.com/psf/black
-    rev: 22.12.0
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v5.0.0
     hooks:
-      - id: black
-        language_version: python3.12
-
-
+      - id: check-added-large-files
+        args: [— maxkb=1000]
+        name: Check for added large files
+      - id: check-ast
+        name: Check Python files for parse errors
+      - id: check-case-conflict
+        name: Check files for case conflicts on case-insensitive filesystems
+      - id: check-docstring-first
+        name: Check that no code is placed prior to docstrings
+      - id: check-executables-have-shebangs
+        name: Check that executables have shebangs
+      - id: check-illegal-windows-names
+        name: Check for illegal Windows filenames
+      - id: check-json
+        name: Check valid JSON syntax
+      - id: check-merge-conflict
+        name: Check for files that contain merge conflict strings
+      - id: check-shebang-scripts-are-executable
+        name: Check that shebang scripts are executable
+      - id: check-toml
+        name: Check valid TOML syntax
+      - id: check-xml
+        name: Check valid XML syntax
+      - id: check-yaml
+        name: Check valid YAML syntax
+      - id: end-of-file-fixer
+        name: Check for newline at end of file
+      - id: debug-statements
+        name: Check for debugger imports
+      - id: no-commit-to-branch
+        name: Check to not commit to main
+      - id: trailing-whitespace
+        name: Check for trailing whitespaces
+  - repo: https://github.com/PyCQA/bandit
+    rev: 1.8.0
+    hooks:
+      - id: bandit
+        name: Run bandit (security linter for python)
+        args: ["-c", "pyproject.toml", "-r", "-ll", "meshpy/"]
+        additional_dependencies: ["bandit[toml]"]
+  - repo: https://github.com/pre-commit/mirrors-clang-format
+    rev: v19.1.5
+    hooks:
+      - id: clang-format
+        name: Run clang-format (C/C++ formatter)
+  - repo: https://github.com/PyCQA/docformatter
+    rev: "eb1df347edd128b30cd3368dddc3aa65edcfac38" # change back to version once bug is fixed in latest version to be compatible with pre-commit
+    hooks:
+      - id: docformatter
+        name: Run docformatter (formatter for docstrings)
+        args: ["--in-place", "--config=./pyproject.toml", "-r", "meshpy/"]
+  - repo: https://github.com/econchick/interrogate
```

Comment:
Very nice, once we have that we can also enforce this in testing with a future PR (I think it should not be too much effort to add all the currently missing docstrings to get to a 100%).

Reply:
Yes - that should also be a separate/future PR

Reply:
Sounds good! Can we somehow easily print for which functions this is missing?

Reply:
Yes - once we increase the percent you automatically get an exhaustive overview on which docstrings are missing